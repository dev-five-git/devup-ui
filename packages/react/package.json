{
  "name": "@devup-ui/react",
  "description": "Zero Config, Zero FOUC, Zero Runtime, CSS in JS Preprocessor",
  "repository": "https://github.com/dev-five-git/devup-ui",
  "author": "devfive",
  "license": "Apache-2.0",
  "homepage": "https://devup-ui.com",
  "bugs": {
    "url": "https://github.com/dev-five-git/devup-ui/issues",
    "email": "contact@devfive.kr"
  },
  "keywords": [
    "css",
    "css-in-js",
    "css-in-js-preprocessor",
    "css-in-js-framework",
    "react"
  ],
  "version": "1.0.24",
  "type": "module",
  "scripts": {
    "lint": "eslint && tsc --noEmit",
    "build": "tsc && vite build"
  },
  "publishConfig": {
    "access": "public"
  },
  "sideEffects": false,
  "main": "./dist/index.cjs",
  "module": "./dist/index.js",
  "exports": {
    ".": {
      "import": "./dist/index.js",
      "require": "./dist/index.cjs"
    }
  },
  "files": [
    "dist"
  ],
  "types": "./dist/index.d.ts",
  "dependencies": {
    "react": "^19.2",
    "csstype": "latest"
  },
  "devDependencies": {
    "rollup-plugin-preserve-directives": "^0.4",
    "vite": "^7.1",
    "vite-plugin-dts": "^4.5",
    "typescript": "^5.9",
    "@types/react": "^19.2",
    "@types/node": "^24.9"
  },
  "peerDependencies": {
    "react": "*",
    "csstype": "*"
  }
}
