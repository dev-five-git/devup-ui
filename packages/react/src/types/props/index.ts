import type { Properties } from 'csstype'

import type { ResponsiveValue } from '../responsive-value'
import type { Merge } from '../utils'
import type { DevupUiBackgroundProps } from './background'
import type { DevupUiBorderProps } from './border'
import type { DevupUiBoxModelProps } from './box-model'
import type { DevupUiBoxSizingProps } from './box-sizing'
import type { DevupUiFlexProps } from './flex'
import type { DevupUiImageProps } from './image'
import type { DevupUiMaskProps } from './mask'
import type { DevupUiMotionPathProps } from './motion-path'
import type { DevupUiPositionProps } from './position'
import type { DevupSelectorProps, DevupThemeSelectorProps } from './selector'
import type { DevupUiTextProps } from './text'

export interface DevupShortcutsProps
  extends DevupUiBackgroundProps,
    DevupUiBorderProps,
    DevupUiBoxModelProps,
    DevupUiBoxSizingProps,
    DevupUiFlexProps,
    DevupUiImageProps,
    DevupUiMotionPathProps,
    DevupUiPositionProps,
    DevupUiMaskProps,
    DevupUiTextProps {}

export type DevupCommonProps = Merge<
  {
    [K in keyof Properties]?: ResponsiveValue<Properties[K]>
  },
  DevupShortcutsProps
>

export interface DevupProps extends DevupCommonProps, DevupSelectorProps {}

export interface DevupPropsWithTheme
  extends DevupProps,
    DevupThemeSelectorProps {}

export interface DevupComponentProps<T extends React.ElementType>
  extends DevupPropsWithTheme {
  as?: T
  styleVars?: Record<string, string | undefined>
}
