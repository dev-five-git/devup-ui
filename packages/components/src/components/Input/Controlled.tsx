import { useState } from 'react'

import { Input } from '.'

export function Controlled() {
  const [value, setValue] = useState('')

  return <Input onChange={(e) => setValue(e.target.value)} value={value} />
}
