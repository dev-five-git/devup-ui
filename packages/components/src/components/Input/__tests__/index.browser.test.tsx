import { fireEvent, render } from '@testing-library/react'
import { DevupThemeTypography } from 'node_modules/@devup-ui/react/dist/types/typography'

import { ClearButton, Input } from '..'
import { Controlled } from '../Controlled'
import { GlassIcon } from '../GlassIcon'

describe('Input', () => {
  it('should render with default props', () => {
    const { container } = render(<Input />)
    expect(container).toMatchSnapshot()
  })

  it('shoue have class name when className prop is provided', () => {
    const { container } = render(
      <Input
        className="test"
        classNames={{
          input: 'test',
          container: 'test',
          icon: 'test',
          errorMessage: 'test',
        }}
      />,
    )
    expect(container).toMatchSnapshot()
    expect(container.querySelector('[aria-label="input"]')).toHaveClass('test')
  })

  it('should render with disabled prop', () => {
    const { container } = render(<Input disabled />)
    expect(container).toMatchSnapshot()
  })

  it('should render with allowClear prop', () => {
    const { container } = render(<Input allowClear />)
    expect(container).toMatchSnapshot()
    expect(container.querySelector('[aria-label="input"]')).toHaveClass(
      'padding-right-0-36px--1',
    )
  })

  it('should not have padding right when allowClear is false', () => {
    const { container } = render(<Input allowClear={false} />)
    expect(container).toMatchSnapshot()
    expect(container.querySelector('[aria-label="input"]')).not.toHaveClass(
      'padding-right-0-36px--1',
    )
  })

  it('should show clear button when value is not empty', () => {
    const { container } = render(<Input />)
    expect(container).toMatchSnapshot()
    fireEvent.change(container.querySelector('input')!, {
      target: { value: 'test' },
    })
    expect(container.querySelector('button')).toBeInTheDocument()
  })

  it('should not show clear button when value is empty', () => {
    const { container } = render(<Input />)
    expect(container).toMatchSnapshot()
  })

  it('should be able to clear value by clicking clear button', () => {
    const { container } = render(<Input allowClear />)
    fireEvent.change(container.querySelector('input')!, {
      target: { value: 'test' },
    })
    expect(container.querySelector('button')).toBeInTheDocument()
    fireEvent.click(container.querySelector('button')!)
    expect(container.querySelector('input')!.value).toBe('')
  })

  it('should be able to render with icon', () => {
    const { container } = render(
      <Input icon={<GlassIcon data-testid="icon" />} />,
    )
    expect(container.querySelector('[data-testid="icon"]')).toBeInTheDocument()
  })

  it('should render error style when error is true', () => {
    const { container } = render(<Input error />)
    expect(container).toMatchSnapshot()
    expect(container.querySelector('[aria-label="input"]')).toHaveClass(
      'border-color-0-var(--error,light-dark(#D52B2E,#FF5B5E))--1',
    )
  })

  it('should be able to render with error message', () => {
    const { container } = render(<Input error errorMessage="Error message" />)
    expect(
      container.querySelector('[aria-label="error-message"]'),
    ).toBeInTheDocument()
  })

  it('should pass colors prop', () => {
    const { container } = render(
      <Input
        colors={{
          primary: 'red',
          error: 'blue',
          text: 'green',
        }}
      />,
    )
    const input = container.querySelector('[aria-label="input"]')
    expect(input).toHaveStyle({
      '--primary': 'red',
      '--error': 'blue',
      '--text': 'green',
    })
  })

  it('should have typography when typography is provided', () => {
    const { container } = render(
      <Input typography={'inlineLabelS' as keyof DevupThemeTypography} />,
    )
    expect(container).toMatchSnapshot()
    expect(container.querySelector('input')).toHaveClass('typo-inlineLabelS')
  })

  it('should pass className prop to error message component', () => {
    const { container } = render(
      <Input
        classNames={{
          errorMessage: 'error-message',
        }}
        error
        errorMessage="Error message"
      />,
    )
    expect(container).toMatchSnapshot()
    expect(container.querySelector('[aria-label="error-message"]')).toHaveClass(
      'error-message',
    )
  })

  it('should pass className prop to icon component', () => {
    const { container } = render(
      <Input
        classNames={{
          icon: 'icon',
        }}
        icon={<GlassIcon />}
      />,
    )
    expect(container).toMatchSnapshot()
    expect(container.querySelector('[aria-label="icon"]')).toHaveClass('icon')
  })

  it('should pass props to ClearButton component', async () => {
    const { container } = render(<ClearButton />)
    expect(container).toMatchSnapshot()
    const clearButton = container.querySelector('[aria-label="clear-button"]')
    expect(clearButton).toBeInTheDocument()
  })

  it('should render disabled icon style when disabled is true', () => {
    const { container } = render(<Input disabled icon={<GlassIcon />} />)
    expect(container).toMatchSnapshot()
    expect(container.querySelector('[aria-label="icon"]')).toHaveClass(
      'color-0-var(--inputDisabledText,light-dark(#D6D7DE,#373737))--1',
    )
  })

  it('should call onChange prop when it is provided andvalue is changed', () => {
    const onChange = vi.fn()
    const { container } = render(<Input onChange={onChange} />)
    fireEvent.change(container.querySelector('input')!, {
      target: { value: 'test' },
    })
    expect(onChange).toHaveBeenCalledWith(expect.any(Object))
  })

  it('should call onClear props when click clear button', () => {
    const onClear = vi.fn()
    const { container } = render(<Input onClear={onClear} />)
    fireEvent.change(container.querySelector('input')!, {
      target: { value: 'test' },
    })
    fireEvent.click(container.querySelector('[aria-label="clear-button"]')!)
    expect(onClear).toHaveBeenCalled()
  })
})

describe('Controlled Input', () => {
  it('should render with value', () => {
    const { container } = render(<Controlled />)
    expect(container).toMatchSnapshot()
  })

  it('should update value when it is changed', () => {
    const { container } = render(<Controlled />)
    fireEvent.change(container.querySelector('input')!, {
      target: { value: 'test' },
    })
    expect(container.querySelector('input')!.value).toBe('test')
  })
})
