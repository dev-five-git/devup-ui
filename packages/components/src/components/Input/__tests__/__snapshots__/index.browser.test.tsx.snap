// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Controlled Input > should render with value 1`] = `
<div>
  <div
    class=" display-0-inline-block--255 position-0-relative--255 box-sizing-0-border-box-1137980104803605758-255"
  >
    <input
      aria-label="input"
      class="undefined color-0-var(--inputDisabledText,light-dark(#D6D7DE,#373737))-7704306251139592248-1 background-0-var(--inputDisabledBg,light-dark(#F0F0F3,#414244))-14172363753176421546-1 border-0-1px solid var(--border,light-dark(#E4E4E4,#434343))-14172363753176421546-1 color-0-var(--inputDisabledText,light-dark(#D6D7DE,#373737))-14172363753176421546-1 background-0-var(--primaryBg,light-dark(#F4F3FA,#F4F3FA0D))-16231282139879731451-1 border-0-1px solid var(--primary,light-dark(#674DC7,#8163E1))-16231282139879731451-1 outline-0-none-16231282139879731451-1 border-0-1px solid var(--primary,light-dark(#674DC7,#8163E1))-8380715471663921674-1 color-0-var(--inputPlaceholder,light-dark(#A9A8AB,#CBCBCB))-15878565022192187906-1 background-0-var(--inputBg,light-dark(#FFF,#2E2E2E))--1 border-radius-0-8px--1 border-style-0-solid--1 border-width-0-1px--1 padding-bottom-0-12px--1 padding-top-0-12px--1 transition-0-all .1s ease-in-out--1 border-color-0-var(--border,light-dark(#E4E4E4,#434343))--1 padding-left-0-12px--1 padding-right-0-36px--1 "
      value=""
    />
  </div>
</div>
`;

exports[`Input > should have typography when typography is provided 1`] = `
<div>
  <div
    class=" display-0-inline-block--255 position-0-relative--255 box-sizing-0-border-box-1137980104803605758-255"
  >
    <input
      aria-label="input"
      class="undefined color-0-var(--inputDisabledText,light-dark(#D6D7DE,#373737))-7704306251139592248-1 background-0-var(--inputDisabledBg,light-dark(#F0F0F3,#414244))-14172363753176421546-1 border-0-1px solid var(--border,light-dark(#E4E4E4,#434343))-14172363753176421546-1 color-0-var(--inputDisabledText,light-dark(#D6D7DE,#373737))-14172363753176421546-1 background-0-var(--primaryBg,light-dark(#F4F3FA,#F4F3FA0D))-16231282139879731451-1 border-0-1px solid var(--primary,light-dark(#674DC7,#8163E1))-16231282139879731451-1 outline-0-none-16231282139879731451-1 border-0-1px solid var(--primary,light-dark(#674DC7,#8163E1))-8380715471663921674-1 color-0-var(--inputPlaceholder,light-dark(#A9A8AB,#CBCBCB))-15878565022192187906-1 background-0-var(--inputBg,light-dark(#FFF,#2E2E2E))--1 border-radius-0-8px--1 border-style-0-solid--1 border-width-0-1px--1 padding-bottom-0-12px--1 padding-top-0-12px--1 transition-0-all .1s ease-in-out--1 border-color-0-var(--border,light-dark(#E4E4E4,#434343))--1 padding-left-0-12px--1 padding-right-0-36px--1 typo-inlineLabelS"
      value=""
    />
  </div>
</div>
`;

exports[`Input > should not have padding right when allowClear is false 1`] = `
<div>
  <div
    class=" display-0-inline-block--255 position-0-relative--255 box-sizing-0-border-box-1137980104803605758-255"
  >
    <input
      aria-label="input"
      class="undefined color-0-var(--inputDisabledText,light-dark(#D6D7DE,#373737))-7704306251139592248-1 background-0-var(--inputDisabledBg,light-dark(#F0F0F3,#414244))-14172363753176421546-1 border-0-1px solid var(--border,light-dark(#E4E4E4,#434343))-14172363753176421546-1 color-0-var(--inputDisabledText,light-dark(#D6D7DE,#373737))-14172363753176421546-1 background-0-var(--primaryBg,light-dark(#F4F3FA,#F4F3FA0D))-16231282139879731451-1 border-0-1px solid var(--primary,light-dark(#674DC7,#8163E1))-16231282139879731451-1 outline-0-none-16231282139879731451-1 border-0-1px solid var(--primary,light-dark(#674DC7,#8163E1))-8380715471663921674-1 color-0-var(--inputPlaceholder,light-dark(#A9A8AB,#CBCBCB))-15878565022192187906-1 background-0-var(--inputBg,light-dark(#FFF,#2E2E2E))--1 border-radius-0-8px--1 border-style-0-solid--1 border-width-0-1px--1 padding-bottom-0-12px--1 padding-top-0-12px--1 transition-0-all .1s ease-in-out--1 border-color-0-var(--border,light-dark(#E4E4E4,#434343))--1 padding-left-0-12px--1 padding-right-0-12px--1 "
      value=""
    />
  </div>
</div>
`;

exports[`Input > should not show clear button when value is empty 1`] = `
<div>
  <div
    class=" display-0-inline-block--255 position-0-relative--255 box-sizing-0-border-box-1137980104803605758-255"
  >
    <input
      aria-label="input"
      class="undefined color-0-var(--inputDisabledText,light-dark(#D6D7DE,#373737))-7704306251139592248-1 background-0-var(--inputDisabledBg,light-dark(#F0F0F3,#414244))-14172363753176421546-1 border-0-1px solid var(--border,light-dark(#E4E4E4,#434343))-14172363753176421546-1 color-0-var(--inputDisabledText,light-dark(#D6D7DE,#373737))-14172363753176421546-1 background-0-var(--primaryBg,light-dark(#F4F3FA,#F4F3FA0D))-16231282139879731451-1 border-0-1px solid var(--primary,light-dark(#674DC7,#8163E1))-16231282139879731451-1 outline-0-none-16231282139879731451-1 border-0-1px solid var(--primary,light-dark(#674DC7,#8163E1))-8380715471663921674-1 color-0-var(--inputPlaceholder,light-dark(#A9A8AB,#CBCBCB))-15878565022192187906-1 background-0-var(--inputBg,light-dark(#FFF,#2E2E2E))--1 border-radius-0-8px--1 border-style-0-solid--1 border-width-0-1px--1 padding-bottom-0-12px--1 padding-top-0-12px--1 transition-0-all .1s ease-in-out--1 border-color-0-var(--border,light-dark(#E4E4E4,#434343))--1 padding-left-0-12px--1 padding-right-0-36px--1 "
      value=""
    />
  </div>
</div>
`;

exports[`Input > should pass className prop to error message component 1`] = `
<div>
  <div
    class=" display-0-inline-block--255 position-0-relative--255 box-sizing-0-border-box-1137980104803605758-255"
  >
    <input
      aria-label="input"
      class="undefined color-0-var(--inputDisabledText,light-dark(#D6D7DE,#373737))-7704306251139592248-1 background-0-var(--inputDisabledBg,light-dark(#F0F0F3,#414244))-14172363753176421546-1 border-0-1px solid var(--border,light-dark(#E4E4E4,#434343))-14172363753176421546-1 color-0-var(--inputDisabledText,light-dark(#D6D7DE,#373737))-14172363753176421546-1 background-0-var(--primaryBg,light-dark(#F4F3FA,#F4F3FA0D))-16231282139879731451-1 border-0-1px solid var(--primary,light-dark(#674DC7,#8163E1))-16231282139879731451-1 outline-0-none-16231282139879731451-1 border-0-1px solid var(--primary,light-dark(#674DC7,#8163E1))-8380715471663921674-1 color-0-var(--inputPlaceholder,light-dark(#A9A8AB,#CBCBCB))-15878565022192187906-1 background-0-var(--inputBg,light-dark(#FFF,#2E2E2E))--1 border-radius-0-8px--1 border-style-0-solid--1 border-width-0-1px--1 padding-bottom-0-12px--1 padding-top-0-12px--1 transition-0-all .1s ease-in-out--1 border-color-0-var(--error,light-dark(#D52B2E,#FF5B5E))--1 padding-left-0-12px--1 padding-right-0-36px--1 "
      value=""
    />
    <span
      aria-label="error-message"
      class="error-message bottom-0--8px--1 color-0-var(--error,light-dark(#D52B2E,#FF5B5E))--1 left-0-0--1 position-0-absolute--1 transform-0-translateY(100%)--1 typo-inputPlaceholder"
    >
      Error message
    </span>
  </div>
</div>
`;

exports[`Input > should pass className prop to icon component 1`] = `
<div>
  <div
    class=" display-0-inline-block--255 position-0-relative--255 box-sizing-0-border-box-1137980104803605758-255"
  >
    <div
      aria-label="icon"
      class="icon display-0-flex--0 justify-content-0-center--0 align-items-0-center--0 height-0-24px--1 width-0-24px--1 left-0-12px--1 position-0-absolute--1 top-0-50%--1 transform-0-translateY(-50%)--1 color-0-var(--iconBold,light-dark(#8D8C9A,#666577))--1"
    >
      <svg
        fill="none"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M14.5006 15.9949C13.445 16.6754 12.1959 17.069 10.8571 17.069C7.07005 17.069 4 13.9195 4 10.0345C4 6.14945 7.07005 3 10.8571 3C14.6442 3 17.7143 6.14945 17.7143 10.0345C17.7143 11.7044 17.1471 13.2384 16.1995 14.4448C16.2121 14.4567 16.2245 14.4688 16.2367 14.4813L19.6653 17.9986C20.1116 18.4564 20.1116 19.1988 19.6653 19.6566C19.2189 20.1145 18.4953 20.1145 18.049 19.6566L14.6204 16.1394C14.5761 16.0938 14.5361 16.0455 14.5006 15.9949ZM16.2143 10.0345C16.2143 13.1274 13.7799 15.569 10.8571 15.569C7.93435 15.569 5.5 13.1274 5.5 10.0345C5.5 6.94154 7.93435 4.5 10.8571 4.5C13.7799 4.5 16.2143 6.94154 16.2143 10.0345Z"
          fill="currentColor"
          fill-rule="evenodd"
        />
      </svg>
    </div>
    <input
      aria-label="input"
      class="undefined color-0-var(--inputDisabledText,light-dark(#D6D7DE,#373737))-7704306251139592248-1 background-0-var(--inputDisabledBg,light-dark(#F0F0F3,#414244))-14172363753176421546-1 border-0-1px solid var(--border,light-dark(#E4E4E4,#434343))-14172363753176421546-1 color-0-var(--inputDisabledText,light-dark(#D6D7DE,#373737))-14172363753176421546-1 background-0-var(--primaryBg,light-dark(#F4F3FA,#F4F3FA0D))-16231282139879731451-1 border-0-1px solid var(--primary,light-dark(#674DC7,#8163E1))-16231282139879731451-1 outline-0-none-16231282139879731451-1 border-0-1px solid var(--primary,light-dark(#674DC7,#8163E1))-8380715471663921674-1 color-0-var(--inputPlaceholder,light-dark(#A9A8AB,#CBCBCB))-15878565022192187906-1 background-0-var(--inputBg,light-dark(#FFF,#2E2E2E))--1 border-radius-0-8px--1 border-style-0-solid--1 border-width-0-1px--1 padding-bottom-0-12px--1 padding-top-0-12px--1 transition-0-all .1s ease-in-out--1 border-color-0-var(--border,light-dark(#E4E4E4,#434343))--1 padding-left-0-36px--1 padding-right-0-36px--1 "
      value=""
    />
  </div>
</div>
`;

exports[`Input > should pass props to ClearButton component 1`] = `
<div>
  <button
    aria-label="clear-button"
    class="align-items-0-center--1 background-0-var(--negative20,light-dark(#0003,#FFF6))--1 border-0-none--1 border-radius-0-50%--1 height-0-20px--1 width-0-20px--1 color-0-var(--base,light-dark(#FFF,#000))--1 cursor-0-pointer--1 display-0-flex--1 justify-content-0-center--1 padding-0-2px--1 position-0-absolute--1 right-0-12px--1 top-0-50%--1 transform-0-translateY(-50%)--1 "
  >
    <svg
      fill="none"
      height="24"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M18 6L6 18"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
      />
      <path
        d="M6 6L18 18"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
      />
    </svg>
  </button>
</div>
`;

exports[`Input > should render disabled icon style when disabled is true 1`] = `
<div>
  <div
    class=" display-0-inline-block--255 position-0-relative--255 box-sizing-0-border-box-1137980104803605758-255"
  >
    <div
      aria-label="icon"
      class=" display-0-flex--0 justify-content-0-center--0 align-items-0-center--0 height-0-24px--1 width-0-24px--1 left-0-12px--1 position-0-absolute--1 top-0-50%--1 transform-0-translateY(-50%)--1 color-0-var(--inputDisabledText,light-dark(#D6D7DE,#373737))--1"
    >
      <svg
        fill="none"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M14.5006 15.9949C13.445 16.6754 12.1959 17.069 10.8571 17.069C7.07005 17.069 4 13.9195 4 10.0345C4 6.14945 7.07005 3 10.8571 3C14.6442 3 17.7143 6.14945 17.7143 10.0345C17.7143 11.7044 17.1471 13.2384 16.1995 14.4448C16.2121 14.4567 16.2245 14.4688 16.2367 14.4813L19.6653 17.9986C20.1116 18.4564 20.1116 19.1988 19.6653 19.6566C19.2189 20.1145 18.4953 20.1145 18.049 19.6566L14.6204 16.1394C14.5761 16.0938 14.5361 16.0455 14.5006 15.9949ZM16.2143 10.0345C16.2143 13.1274 13.7799 15.569 10.8571 15.569C7.93435 15.569 5.5 13.1274 5.5 10.0345C5.5 6.94154 7.93435 4.5 10.8571 4.5C13.7799 4.5 16.2143 6.94154 16.2143 10.0345Z"
          fill="currentColor"
          fill-rule="evenodd"
        />
      </svg>
    </div>
    <input
      aria-label="input"
      class="undefined color-0-var(--inputDisabledText,light-dark(#D6D7DE,#373737))-7704306251139592248-1 background-0-var(--inputDisabledBg,light-dark(#F0F0F3,#414244))-14172363753176421546-1 border-0-1px solid var(--border,light-dark(#E4E4E4,#434343))-14172363753176421546-1 color-0-var(--inputDisabledText,light-dark(#D6D7DE,#373737))-14172363753176421546-1 background-0-var(--primaryBg,light-dark(#F4F3FA,#F4F3FA0D))-16231282139879731451-1 border-0-1px solid var(--primary,light-dark(#674DC7,#8163E1))-16231282139879731451-1 outline-0-none-16231282139879731451-1 border-0-1px solid var(--primary,light-dark(#674DC7,#8163E1))-8380715471663921674-1 color-0-var(--inputPlaceholder,light-dark(#A9A8AB,#CBCBCB))-15878565022192187906-1 background-0-var(--inputBg,light-dark(#FFF,#2E2E2E))--1 border-radius-0-8px--1 border-style-0-solid--1 border-width-0-1px--1 padding-bottom-0-12px--1 padding-top-0-12px--1 transition-0-all .1s ease-in-out--1 border-color-0-var(--border,light-dark(#E4E4E4,#434343))--1 padding-left-0-36px--1 padding-right-0-36px--1 "
      disabled=""
      value=""
    />
  </div>
</div>
`;

exports[`Input > should render error style when error is true 1`] = `
<div>
  <div
    class=" display-0-inline-block--255 position-0-relative--255 box-sizing-0-border-box-1137980104803605758-255"
  >
    <input
      aria-label="input"
      class="undefined color-0-var(--inputDisabledText,light-dark(#D6D7DE,#373737))-7704306251139592248-1 background-0-var(--inputDisabledBg,light-dark(#F0F0F3,#414244))-14172363753176421546-1 border-0-1px solid var(--border,light-dark(#E4E4E4,#434343))-14172363753176421546-1 color-0-var(--inputDisabledText,light-dark(#D6D7DE,#373737))-14172363753176421546-1 background-0-var(--primaryBg,light-dark(#F4F3FA,#F4F3FA0D))-16231282139879731451-1 border-0-1px solid var(--primary,light-dark(#674DC7,#8163E1))-16231282139879731451-1 outline-0-none-16231282139879731451-1 border-0-1px solid var(--primary,light-dark(#674DC7,#8163E1))-8380715471663921674-1 color-0-var(--inputPlaceholder,light-dark(#A9A8AB,#CBCBCB))-15878565022192187906-1 background-0-var(--inputBg,light-dark(#FFF,#2E2E2E))--1 border-radius-0-8px--1 border-style-0-solid--1 border-width-0-1px--1 padding-bottom-0-12px--1 padding-top-0-12px--1 transition-0-all .1s ease-in-out--1 border-color-0-var(--error,light-dark(#D52B2E,#FF5B5E))--1 padding-left-0-12px--1 padding-right-0-36px--1 "
      value=""
    />
  </div>
</div>
`;

exports[`Input > should render with allowClear prop 1`] = `
<div>
  <div
    class=" display-0-inline-block--255 position-0-relative--255 box-sizing-0-border-box-1137980104803605758-255"
  >
    <input
      aria-label="input"
      class="undefined color-0-var(--inputDisabledText,light-dark(#D6D7DE,#373737))-7704306251139592248-1 background-0-var(--inputDisabledBg,light-dark(#F0F0F3,#414244))-14172363753176421546-1 border-0-1px solid var(--border,light-dark(#E4E4E4,#434343))-14172363753176421546-1 color-0-var(--inputDisabledText,light-dark(#D6D7DE,#373737))-14172363753176421546-1 background-0-var(--primaryBg,light-dark(#F4F3FA,#F4F3FA0D))-16231282139879731451-1 border-0-1px solid var(--primary,light-dark(#674DC7,#8163E1))-16231282139879731451-1 outline-0-none-16231282139879731451-1 border-0-1px solid var(--primary,light-dark(#674DC7,#8163E1))-8380715471663921674-1 color-0-var(--inputPlaceholder,light-dark(#A9A8AB,#CBCBCB))-15878565022192187906-1 background-0-var(--inputBg,light-dark(#FFF,#2E2E2E))--1 border-radius-0-8px--1 border-style-0-solid--1 border-width-0-1px--1 padding-bottom-0-12px--1 padding-top-0-12px--1 transition-0-all .1s ease-in-out--1 border-color-0-var(--border,light-dark(#E4E4E4,#434343))--1 padding-left-0-12px--1 padding-right-0-36px--1 "
      value=""
    />
  </div>
</div>
`;

exports[`Input > should render with default props 1`] = `
<div>
  <div
    class=" display-0-inline-block--255 position-0-relative--255 box-sizing-0-border-box-1137980104803605758-255"
  >
    <input
      aria-label="input"
      class="undefined color-0-var(--inputDisabledText,light-dark(#D6D7DE,#373737))-7704306251139592248-1 background-0-var(--inputDisabledBg,light-dark(#F0F0F3,#414244))-14172363753176421546-1 border-0-1px solid var(--border,light-dark(#E4E4E4,#434343))-14172363753176421546-1 color-0-var(--inputDisabledText,light-dark(#D6D7DE,#373737))-14172363753176421546-1 background-0-var(--primaryBg,light-dark(#F4F3FA,#F4F3FA0D))-16231282139879731451-1 border-0-1px solid var(--primary,light-dark(#674DC7,#8163E1))-16231282139879731451-1 outline-0-none-16231282139879731451-1 border-0-1px solid var(--primary,light-dark(#674DC7,#8163E1))-8380715471663921674-1 color-0-var(--inputPlaceholder,light-dark(#A9A8AB,#CBCBCB))-15878565022192187906-1 background-0-var(--inputBg,light-dark(#FFF,#2E2E2E))--1 border-radius-0-8px--1 border-style-0-solid--1 border-width-0-1px--1 padding-bottom-0-12px--1 padding-top-0-12px--1 transition-0-all .1s ease-in-out--1 border-color-0-var(--border,light-dark(#E4E4E4,#434343))--1 padding-left-0-12px--1 padding-right-0-36px--1 "
      value=""
    />
  </div>
</div>
`;

exports[`Input > should render with disabled prop 1`] = `
<div>
  <div
    class=" display-0-inline-block--255 position-0-relative--255 box-sizing-0-border-box-1137980104803605758-255"
  >
    <input
      aria-label="input"
      class="undefined color-0-var(--inputDisabledText,light-dark(#D6D7DE,#373737))-7704306251139592248-1 background-0-var(--inputDisabledBg,light-dark(#F0F0F3,#414244))-14172363753176421546-1 border-0-1px solid var(--border,light-dark(#E4E4E4,#434343))-14172363753176421546-1 color-0-var(--inputDisabledText,light-dark(#D6D7DE,#373737))-14172363753176421546-1 background-0-var(--primaryBg,light-dark(#F4F3FA,#F4F3FA0D))-16231282139879731451-1 border-0-1px solid var(--primary,light-dark(#674DC7,#8163E1))-16231282139879731451-1 outline-0-none-16231282139879731451-1 border-0-1px solid var(--primary,light-dark(#674DC7,#8163E1))-8380715471663921674-1 color-0-var(--inputPlaceholder,light-dark(#A9A8AB,#CBCBCB))-15878565022192187906-1 background-0-var(--inputBg,light-dark(#FFF,#2E2E2E))--1 border-radius-0-8px--1 border-style-0-solid--1 border-width-0-1px--1 padding-bottom-0-12px--1 padding-top-0-12px--1 transition-0-all .1s ease-in-out--1 border-color-0-var(--border,light-dark(#E4E4E4,#434343))--1 padding-left-0-12px--1 padding-right-0-36px--1 "
      disabled=""
      value=""
    />
  </div>
</div>
`;

exports[`Input > should show clear button when value is not empty 1`] = `
<div>
  <div
    class=" display-0-inline-block--255 position-0-relative--255 box-sizing-0-border-box-1137980104803605758-255"
  >
    <input
      aria-label="input"
      class="undefined color-0-var(--inputDisabledText,light-dark(#D6D7DE,#373737))-7704306251139592248-1 background-0-var(--inputDisabledBg,light-dark(#F0F0F3,#414244))-14172363753176421546-1 border-0-1px solid var(--border,light-dark(#E4E4E4,#434343))-14172363753176421546-1 color-0-var(--inputDisabledText,light-dark(#D6D7DE,#373737))-14172363753176421546-1 background-0-var(--primaryBg,light-dark(#F4F3FA,#F4F3FA0D))-16231282139879731451-1 border-0-1px solid var(--primary,light-dark(#674DC7,#8163E1))-16231282139879731451-1 outline-0-none-16231282139879731451-1 border-0-1px solid var(--primary,light-dark(#674DC7,#8163E1))-8380715471663921674-1 color-0-var(--inputPlaceholder,light-dark(#A9A8AB,#CBCBCB))-15878565022192187906-1 background-0-var(--inputBg,light-dark(#FFF,#2E2E2E))--1 border-radius-0-8px--1 border-style-0-solid--1 border-width-0-1px--1 padding-bottom-0-12px--1 padding-top-0-12px--1 transition-0-all .1s ease-in-out--1 border-color-0-var(--border,light-dark(#E4E4E4,#434343))--1 padding-left-0-12px--1 padding-right-0-36px--1 "
      value=""
    />
  </div>
</div>
`;
